File: Programming Basics with Python/Вложени цикли/travelling.py


destination = input()
budget = int(input())
sum_input = int(input())

sum = 0
while destination != "End":
    sum += sum_input
    if sum >= budget:
        print(f"Going to {destination}!")
        sum = 0
        destination = input()
        if destination == "End":
            break

        budget = int(input())
        sum_input = int(input())
    else:
        sum_input = int(input())

File: Programming Basics with Java/Nested Loops/Travelling.java
import java.util.Scanner;

public class Travelling 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);

        String destination = input.nextLine();
        double trip_savings = 0;

        while (!destination.equals("End"))
        {
            double budget = Double.parseDouble(input.nextLine());

            while(trip_savings < budget)
            {
                trip_savings += Double.parseDouble(input.nextLine());
            }

            System.out.printf("Going to %s!\n" , destination);
            trip_savings = 0;
            destination = input.nextLine();
        }
    }
}
